dot/core/messages.go

```diff
@@ -29,18 +28,26 @@ func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt Run
 	rt.SetContextStorage(ts)
 
 	// validate each transaction
-	externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
+	externalExt, err := s.buildExternalTransaction(rt, tx)
+	if err != nil {
+		return nil, false, fmt.Errorf("unable to build transaction: %s", err)
```

Comment:
nit to unclutter `unable to` repetition with errors coming from a nested call

```suggestion
		return nil, false, fmt.Errorf("building transaction: %s", err)
```